File: socialNetwork/src/main/java/com/ftn/socialNetwork/model/entity/Reaction.java
package com.ftn.socialNetwork.model.entity;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "reactions")
public class Reaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private EReactionType type;


    @Column(nullable = false, columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP")
    private LocalDate timeStamp;

    @ManyToOne
    private User user;
}

File: src/app/reaction/reaction.ts
import { User } from "../user";
import { Post } from "../post/post";
import { Comment } from "../comment/comment";
import { EReactionType } from "./eReactionType";

export class Reaction {
  id: number;
  type: EReactionType;
  timeStamp: Date;
  user: User;
  post: Post;
  comment: Comment;

  constructor(
    id: number,
    type: EReactionType,
    timeStamp: Date,
    user: User,
    post: Post,
    comment: Comment
  ) {
    this.id = id;
    this.type = type;
    this.timeStamp = timeStamp;
    this.user = user;
    this.post = post;
    this.comment = comment;
  }
}
